src/app/services/data.service.ts

```diff
@@ -215,16 +261,19 @@ export class DataService {
         // Calculate availability of this component
         this.calculateAvailability(componentId);
       });
-      // We are now fully loaded and can display the data
-      this._loadingFinished.next(true);
+      // Check if we are now fully loaded and can display the data
+      componentsDone = true;
+      if (componentsDone && updatesDone) {
+        this._loadingFinished.next(true);
+      }
     });
   }
 
   private calculateAvailability(component: ComponentId): void {
     const statusList = this.componentStatusByDay.get(component);
     if (!statusList) {
       // TODO Error properly
```

Comment:
Use toastr?

Reply:
Removed the comment. I'm not sure if using Toastr in the data service is such a good idea, given that it might run before the Toastr service has been initialized.